File: leetcode/해시맵 1. Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int cur = nums[i];
            if (map.containsKey(target - cur)) {
                int[] ret = {map.get(target - cur), i};
                return ret;
            }
            map.put(cur, i);
        }
        return null;
    }
}

File: leetcode/해시맵 1. Two Sum.py
class Solution:
    def twoSum(self, nums, target):
        d = dict()
        for i, cur in enumerate(nums):
            if target - cur in d:
                return [d[target - cur], i]
            d[cur] = i


solution = Solution()
print(solution.twoSum([2, 7, 11, 15], 9))
